test/gtest/ucp/test_ucp_proto_mock.cc

```diff
@@ -8,130 +8,80 @@
+    using perf_attr_func_t  = void (*)(uct_perf_attr_t&);
+    using iface_attr_func_t = void (*)(uct_iface_attr&);
 
-    static ucs::mock &mock()
+    mock_iface()
     {
-        static ucs::mock instance;
-        return instance;
+        ucs_assert(singleton == nullptr);
+        singleton = this;
     }
 
-    static mock_component *find(uct_component_h component)
+    ~mock_iface()
     {
-        auto it = map().find(component);
-        return (it == map().end()) ? nullptr : it->second.get();
+        cleanup();
+        singleton = nullptr;
     }
 
-    static uct_component_h find_uct_component(const std::string &name)
+    void cleanup()
     {
-        uct_component_h component;
-        ucs_list_for_each(component, &uct_components_list, list) {
-            if (name == component->name) {
-                return component;
-            }
-        }
-
-        return nullptr;
+        m_mock.cleanup();
     }
-};
 
-class mock_component {
-public:
-    /* We could use std::function here, but coverity complains on std::function
-     * copy ctor */
-    using perf_attr_func_t  = void (*)(uct_perf_attr_t&);
-    using iface_attr_func_t = void (*)(uct_iface_attr&);
-
-    void set_mock_perf_attr(const std::string &iface_name, perf_attr_func_t cb)
+    void
+    set_mock_perf_attr(const std::string &tl_name, perf_attr_func_t cb)
     {
-        m_perf_attrs_funcs[iface_name] = cb;
+        mock_tl(tl_name);
+        m_perf_attrs_funcs[tl_name] = cb;
     }
 
     void
-    set_mock_iface_attr(const std::string &iface_name, iface_attr_func_t cb)
+    set_mock_iface_attr(const std::string &tl_name, iface_attr_func_t cb)
     {
-        m_iface_attrs_funcs[iface_name] = cb;
+        mock_tl(tl_name);
+        m_iface_attrs_funcs[tl_name] = cb;
     }
 
 private:
-    friend class mock_component_test;
-
-    mock_component(uct_component_h component, ucs::mock &mock) :
-        m_component(component), m_mock(mock)
+    void mock_tl(const std::string &tl_name)
     {
-        uct_tl_t *tl;
-        ucs_list_for_each(tl, &component->tl_list, list) {
-            m_mock.setup(&tl->iface_open, iface_open_mock);
-        }
-    }
-
-    /* Non-copyable, non-moveable */
-    mock_component(const mock_component &) = delete;
-    mock_component(mock_component &&)      = delete;
-    mock_component & operator=(const mock_component &) = delete;
-    mock_component & operator=(mock_component &&)      = delete;
-
-    uct_tl_t *find_tl(const uct_iface_params_t *params) const
-    {
-        uct_tl_t *tl;
-        ucs_list_for_each(tl, &m_component->tl_list, list) {
-            if (0 == strcmp(tl->name, params->mode.device.tl_name)) {
-                return tl;
+        uct_component_h component;
+        ucs_list_for_each(component, &uct_components_list, list) {
+            uct_tl_t *tl;
+            ucs_list_for_each(tl, &component->tl_list, list) {
+                if (tl_name == tl->name) {
+                    EXPECT_EQ(0, m_tls.count(tl_name));
```

Comment:
why needed?

Reply:
Currently mock works with a single transport, so this is just a check that we don't try to mock multiple ones. In the new impl it's slightly different